pkg/webhook/ephemeraljob/validating/validation.go

```diff
@@ -0,0 +1,246 @@
+var supportedPortProtocols = sets.NewString(string(core.ProtocolTCP), string(core.ProtocolUDP), string(core.ProtocolSCTP))
+
+var allowedEphemeralContainerFields = map[string]bool{
+	"Name":                     true,
+	"Image":                    true,
+	"Command":                  true,
+	"Args":                     true,
+	"WorkingDir":               true,
+	"Ports":                    false,
+	"EnvFrom":                  true,
+	"Env":                      true,
+	"Resources":                false,
+	"VolumeMounts":             true,
+	"VolumeDevices":            true,
+	"LivenessProbe":            false,
+	"ReadinessProbe":           false,
+	"StartupProbe":             false,
+	"Lifecycle":                false,
+	"TerminationMessagePath":   true,
+	"TerminationMessagePolicy": true,
+	"ImagePullPolicy":          true,
+	"SecurityContext":          true,
+	"Stdin":                    true,
+	"StdinOnce":                true,
+	"TTY":                      true,
+}
+
+func validateContainerPorts(ports []core.ContainerPort, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allNames := sets.String{}
+	for i, port := range ports {
+		idxPath := fldPath.Index(i)
+		if len(port.Name) > 0 {
+			if msgs := utilvalidation.IsValidPortName(port.Name); len(msgs) != 0 {
+				for i = range msgs {
+					allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), port.Name, msgs[i]))
+				}
+			} else if allNames.Has(port.Name) {
+				allErrs = append(allErrs, field.Duplicate(idxPath.Child("name"), port.Name))
+			} else {
+				allNames.Insert(port.Name)
+			}
+		}
+		if port.ContainerPort == 0 {
+			allErrs = append(allErrs, field.Required(idxPath.Child("containerPort"), ""))
+		} else {
+			for _, msg := range utilvalidation.IsValidPortNum(int(port.ContainerPort)) {
+				allErrs = append(allErrs, field.Invalid(idxPath.Child("containerPort"), port.ContainerPort, msg))
+			}
+		}
+		if port.HostPort != 0 {
+			for _, msg := range utilvalidation.IsValidPortNum(int(port.HostPort)) {
+				allErrs = append(allErrs, field.Invalid(idxPath.Child("hostPort"), port.HostPort, msg))
+			}
+		}
+		if len(port.Protocol) == 0 {
+			allErrs = append(allErrs, field.Required(idxPath.Child("protocol"), ""))
+		} else if !supportedPortProtocols.Has(string(port.Protocol)) {
+			allErrs = append(allErrs, field.NotSupported(idxPath.Child("protocol"), port.Protocol, supportedPortProtocols.List()))
+		}
+	}
+	return allErrs
+}
+
+func validatePullPolicy(policy core.PullPolicy, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+
+	switch policy {
+	case core.PullAlways, core.PullIfNotPresent, core.PullNever:
+		break
+	case "":
+		allErrors = append(allErrors, field.Required(fldPath, ""))
+	default:
+		allErrors = append(allErrors, field.NotSupported(fldPath, policy, supportedPullPolicies.List()))
+	}
+	return allErrors
+}
+
+func validateResizePolicy(policyList []core.ContainerResizePolicy, fldPath *field.Path) field.ErrorList {
```

Comment:
Is it still necessary to resize the temporary container?

Reply:
you're right.